src/commands/publish/package.rs

```diff
@@ -6,18 +6,18 @@ use serde::{self, Deserialize};
 #[derive(Debug, Deserialize)]
 pub struct Package {
     #[serde(default)]
-    main: String,
+    main: PathBuf,
 }
 impl Package {
-    pub fn main(&self, build_dir: &PathBuf) -> Result<String, failure::Error> {
-        if self.main == "" {
+    pub fn main(&self, build_dir: &PathBuf) -> Result<PathBuf, failure::Error> {
+        if self.main == PathBuf::from("") {
             failure::bail!(
                 "The `main` key in your `package.json` file is required; please specified the entrypoint of your Worker.",
```

Comment:
nit: not in this PR but since we're here - can we change `specified` to `specify`